### PR TITLE
Split advanced container configuration article

### PR DESCRIPTION
**Pull request by wouterj**
| Q | A |
| --- | --- |
| Doc fixes? | yes |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | #4228 (partially, 2 and 3) |

### CODE REVIEW

File: components/dependency_injection/synthetic_services.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            // ...
+            $this->container->set('kernel', $this);
+
+            // ...
+        }
+    }
+
+The ``kernel`` service is called a synthetic service. This service has to be
+configured in the container, so the container knows the service does exists
+during compilation (otherwise, services dependening on this ``kernel`` service
```
Comment (stof) [inline components/dependency_injection/synthetic_services.rst]: typo: ``depending``

File: components/dependency_injection/synthetic_services.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            // ...
+            $this->container->set('kernel', $this);
+
+            // ...
+        }
+    }
+
+The ``kernel`` service is called a synthetic service. This service has to be
+configured in the container, so the container knows the service does exists
```
Comment (stof) [inline components/dependency_injection/synthetic_services.rst]: typo: ``exist``

File: components/dependency_injection/definitions.rst [hunk 3]
```diff
@@ -125,3 +125,15 @@ You can also replace any existing method calls with an array of new ones with::
...
     the container see :doc:`/components/dependency_injection/compilation`.
+
+Requiring Files
+~~~~~~~~~~~~~~~
+
+There might be use cases when you need to include another file just before
+the service itself gets loaded. To do so, you can use the
+:method:`Symfony\\Component\\DependencyInjection\\Definition::setFile` method::
+
+    $definition->setFile('/src/path/to/file/foo.php');
```
Comment (stof) [inline components/dependency_injection/definitions.rst]: what about showing the other formats too, as done previously ? Otherwise, we are loosing some content

Comment (WouterJ) [inline components/dependency_injection/definitions.rst]: This is something we have to think about. Currently, we only show PHP formats in this article. However, this article in general looks like it should recieve a face-lift. Let's do it in another PR :)

**Comment by xabbuh** [comment]
:+1:

**Comment by weaverryan** [comment]
Very cool. Let's face-lift the "definitions" entry too if you see fit (per Stof's comment)
